### DISCUSSION TITLE
Mastodon 4.2.10 and MastoUnexpectedError: The server returned data with an unknown encoding. The server may be down.

### DISCUSSION THREAD

**Discussion by GerritKuilder** [neet/masto.js#1144] [Q&A]
Hi All,

just started a new project and also upgraded my mastodon server, the server is up and running, and doing fine but the new project return the above error when running the following:

```
 const client =  await  createRestAPIClient({
        url: "https://mastodon.ozioso.online",
        accessToken: "XXXX"
    });
    const me = await client.v1.accounts.verifyCredentials();
    console.log(me.id)
    console.log(me.username)
```
Am I missing something or can I get more details of the error?

Edit:

it seems to be the stream user that is failing, also failing for my other projects:
 const stream = await client.v1.stream.streamUser();

Logs
```
echNews
↑ POST https://mastodon.ozioso.online/api/v1/stream/stream_user
        body { encoding: 'json', body: undefined }
HTTP failed Response {
  status: 404,
  statusText: 'Not Found',
  headers: Headers {
    server: 'openresty',
    date: 'Mon, 08 Jul 2024 08:45:21 GMT',
    'content-type': '*/*; charset=utf-8',
    'transfer-encoding': 'chunked',
    connection: 'keep-alive',
```

Kind regards,

Gerrit Kuilder

**Comment by neet** [comment]
Hi @GerritKuilder, As of Masto.js v6, we have refined the method to subscribe to a streaming API to more performant and intuitive design. Here is the latest example:
https://github.com/neet/masto.js/blob/main/examples/timeline-with-streaming.ts

If you want to subscribe to a user, you can code as follows:

```ts
import { createStreamingAPIClient } from "masto";

const streaming = createStreamingAPIClient({
  // If you don't know what streaming API URL is, refer /api/v2/instance of your instance
  streamingApiUrl: "<STREAMING API URL>",
  accessToken: "<TOKEN>",
});

const events = streaming.user.subscribe();

for await (const event of events) {
  switch (event.event) {
    case "update": {
      console.log("new post", event.payload.content);
      break;
    }
    case "delete": {
      console.log("deleted post", event.payload);
      break;
    }
    // of course, you can consume other types of events as well
    default: {
      break;
    }
  }
}
```

If you still have a problem, feel free to write a comment here.

**Reply by GerritKuilder** [reply]
Hi Neet,

thank you for your reply.

I am now getting the error:
```
node_modules/masto/dist/index.js:570
            throw new MastoUnexpectedError(`Unknown action type ${action.type}`);
                  ^

MastoUnexpectedError: Unknown action type user
    at WebSocketActionDispatcher.dispatch (file:///home/gerrit/workspace/code/de
```

My old (5.x) code works on 1 user but not on a newly created user, the new 6.x code fails for both users access tokens.

Strange. 

Kind Regards,

Gerrit Kuilder

**Reply by neet** [reply]
Hi @GerritKuilder, sorry for the incorrect information. The proper way to subscribe to the user events is as follows:

```ts
import { createStreamingAPIClient } from "masto";

const streaming = createStreamingAPIClient({
  streamingApiUrl: "<STREAMING API URL>",
  accessToken: "<TOKEN>",
});

const events = streaming.user.subscribe();

for await (const event of events) {
  console.log(event)
}
```

p.s. if you want to catch up with the latest document, I'd recommend you to see the docs:
https://neet.github.io/masto.js/

Also, we have an end-to-end testing for every APIs which you can refer in the tests directory:
https://github.com/neet/masto.js/blob/0d07a2bb38fbd1f94f021f76fda9b08e4c5ee545/tests/streaming/timelines.spec.ts#L165-L168

**Reply by GerritKuilder** [reply]
Hi neet,

thank you for your input, it was just a bad week for coding for me, I usually read docs and try a few things first. I had read the tests but I missed the finer details.

More testing is needed before I will upgrade all my 5.x code to 6.x
Consider this as closed.

Again, thank you for the input and the NPM.

Gerrit Kuilder
